### PR TITLE
feat: Add the apisix-ingress-controller plugin for kubernetes.

### PR DESCRIPTION
**Pull request by sober-wang**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#73 

___
### New feature or improvement
- Describe the details and related test reports.

You can used the plugin quickly show the apisix routes and upstreams in a kubernetes cluster.

### CODE REVIEW

**Comment by tao12345666333** [comment]
Thanks! Good job!

Can you please rebase your PR from master?

File: cmd/plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# apisix-ingress-controller plugin
+
+The apisix-ingress-controoler plugin can quikly show the apisix routes and upstreams in kubernetes cluster.
+
+# Quikly start 
+
+```shell
+git clone http://github.com/apache/apisix-ingress-controller 
+cd cmd/plugin
+sh install-plugin.sh
```
Comment (tokers) [inline cmd/plugin/README.md:10]: Why not assign the executable permission?

File: cmd/plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+```
+
+```shell 
+# Show your routes in kubernetes.
+kubectl apisix-ingress-controller routes -n <APISIX_INGRESS_NS>
```
Comment (tokers) [inline cmd/plugin/README.md:15]: IMHO `apisix-ingress-controller` is too long, maybe `ingress-apisix` is more concise.

File: cmd/plugin/commands/routes/routes.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package routes
+
+import (
+	"strings"
+
+	"github.com/apache/apisix-ingress-controller/cmd/plugin/common"
+	"github.com/apache/apisix-ingress-controller/cmd/plugin/kubectl"
+	"github.com/apache/apisix-ingress-controller/pkg/log"
```
Comment (tokers) [inline cmd/plugin/commands/routes/routes.go:22]: Please sort the imports, we should put the 3rd parties in the middle, and put our own packages at the bottom.

File: cmd/plugin/commands/routes/routes.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	jsoniter "github.com/json-iterator/go"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
```
Comment (tokers) [inline cmd/plugin/commands/routes/routes.go:27]: Add some comments for exportable functions.

File: cmd/plugin/commands/routes/routes.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	for {
+		if common.CheckPort() {
+			log.Info("the k8s port-forward is ready")
+			break
+		}
+		log.Error("the k8s port-forward is not ready")
+	}
+	var reqApisix common.RequestConf
+	reqApisix.AdminKey = pconf.Cfg.APISIX.AdminKey
+	reqApisix.URL = "http://127.0.0.1:9180/apisix/admin"
```
Comment (tokers) [inline cmd/plugin/commands/routes/routes.go:70]: Cannot assume the APISIX address is just `127.0.0.1:9180`.

Comment (tokers) [inline cmd/plugin/commands/routes/routes.go:70]: Also, we have a full-featured APISIX Go SDK, we don't have to create another one here. See `pkg/apisix` for more details.

Comment (sober-wang) [reply cmd/plugin/commands/routes/routes.go:70]: no, I browse the `pkg/apisix` so the code can't resued. I can only rewrite.

Comment (sober-wang) [reply cmd/plugin/commands/routes/routes.go:70]: Beacuse I used the kubernetes port-forward feature in the machine on which this command is run. So just visit the `127.0.0.1:9180`

File: cmd/plugin/README.md (hunk 2 above)
Comment (sober-wang) [reply cmd/plugin/README.md:15]: no, I think our project is `apache/apisix-ingress-controller` , so I think used it.

File: cmd/plugin/README.md (hunk 1 above)
Comment (sober-wang) [reply cmd/plugin/README.md:10]: oh, right. I add the execcutable permission.

**Review by sober-wang — COMMENTED** [review]
beacuse . I used sh executable the install-plugin.sh file. so I don't need executable permission.

File: cmd/plugin/README.md (hunk 1 above)
Comment (sober-wang) [reply cmd/plugin/README.md:10]: beacuse . I used sh executable the install-plugin.sh file. so I don't need executable permission.

File: cmd/plugin/commands/routes/routes.go (hunk 3 above)
Comment (sober-wang) [reply cmd/plugin/commands/routes/routes.go:22]: beacuse the project not used sort the imports.

File: cmd/plugin/commands/routes/routes.go (hunk 5 above)
Comment (tokers) [reply cmd/plugin/commands/routes/routes.go:70]: It's not reasonable to specify the port, we cannot sure whether this port is already in use.

Comment (tokers) [reply cmd/plugin/commands/routes/routes.go:70]: Why?

**Comment by tokers** [comment]
> beacuse . I used sh executable the install-plugin.sh file. so I don't need executable permission.

I mean, add the shebang line, let users run the scripts like `./scripts.sh`.

**Comment by tao12345666333** [comment]
Can you resolve the conflicts? Thanks!

File: docs/en/latest/concepts/annotations.md [hunk 6]
```diff
@@ -1,133 +0,0 @@
----
```
Comment (tao12345666333) [inline docs/en/latest/concepts/annotations.md:1]: Why do you want to delete these documents?

Comment (sober-wang) [reply docs/en/latest/concepts/annotations.md:1]: > Can you resolve the conflicts? Thanks!

Sorry, I'm afraid I don't know how to operate it. Can you help me?

**Comment by sober-wang** [comment]
I can't fix the conflicts. so I will close the PR next resubmit.

File: docs/en/latest/concepts/annotations.md (hunk 6 above)
Comment (tao12345666333) [reply docs/en/latest/concepts/annotations.md]: @sober-wang If you need some help, please ping me.  I have too much news, easy to miss.

**Comment by tao12345666333** [comment]
> I can't fix the conflicts. so I will close the PR next resubmit.

Looking forward to your PR! Thanks!

**Pull request closed without merging**